### PR TITLE
Toolbar button styling

### PR DESCRIPTION
**Pull request by ciar4n**
Implementing... https://magazine.joomla.org/issues/issue-nov-2017/item/3289-episode-iv-a-new-user-interface-for-the-joomla-backend

Toolbar buttons...

image

### CODE REVIEW

**Comment by brianteeman** [comment]
Some buttons didnt appear unless they are hovered 
applicationframehost_2018-02-20_07-40-20

File: package.json [hunk 1]
```diff
@@ -60,7 +61,7 @@
     "karma-jasmine-ajax": "latest",
     "karma-requirejs": "latest",
     "karma-verbose-reporter": "latest",
-    "node-sass": "^4.7.2",
+    "node-sass": "^4.5.3",
```
Comment (C-Lodder) [inline package.json]: Can you please revert this back to 4.7.2

File: package.json [hunk 2]
```diff
@@ -37,14 +37,15 @@
     "@polymer/test-fixture": "latest",
     "@webcomponents/custom-elements": "latest",
     "@webcomponents/webcomponentsjs": "latest",
-    "autoprefixer": "^8.0.0",
+    "autoprefixer": "^7.1.6",
```
Comment (C-Lodder) [inline package.json]: Please also revert this and recompile. You'll notice a load of additional CSS in your commit which is due to additional vendor prefixes, which v8.0.0 removes

File: libraries/src/Service/Provider/Config.php [hunk 3]
```diff
@@ -39,7 +39,7 @@ function (Container $container)
 				{
 					if (!file_exists(JPATH_CONFIGURATION . '/configuration.php'))
 					{
-						return new Registry;
+						return [];
```
Comment (C-Lodder) [inline libraries/src/Service/Provider/Config.php]: Please revert this

File: installation/src/Service/Provider/Application.php [hunk 4]
```diff
@@ -41,7 +41,15 @@ public function register(Container $container)
 			InstallationApplication::class,
 			function (Container $container)
 			{
-				$app = new InstallationApplication(null, $container->get('config'), null, $container);
+				$config = null;
```
Comment (C-Lodder) [inline installation/src/Service/Provider/Application.php]: Revert this

**Comment by ciar4n** [comment]
@brianteeman I can not replicate this (Chrome and FireFox). Would you be loading some cached CSS?

**Comment by brianteeman** [comment]
I dont think so - but I am just trying a complete git pull instead of using patchtester to see as I have had some issues with patchtester when there are a lot of files

**Comment by brianteeman** [comment]
ok - a clean git pull and it works

The buttons look very big to me - especially on a screen like article manager when you have so many 

Can you check the colours used as my a11y checker is saying that the contrast between foreground and background colors meets WCAG 2 AA contrast ratio thresholds

**Comment by brianteeman** [comment]
Not sure if this is supposed to be covered by this PR but the onhover on these subitems is very hard to see

chrome_2018-02-20_09-01-53

**Comment by ciar4n** [comment]
> The buttons look very big to me - especially on a screen like article manager when you have so many

I'm under strict instructions to follow the design so not my call 😄 .. tagging @coolcat-creations @svom 

> Can you check the colours used as my a11y checker is saying that the contrast between foreground and background colors meets WCAG 2 AA contrast ratio thresholds

Just waiting on a color palette which will be applied in a second PR.. once again tagging @coolcat-creations @svom

**Comment by C-Lodder** [comment]
Buttons looks big but once they've been grouped as per the design, it will hopefully be better

**Comment by C-Lodder** [comment]
@brianteeman **WCAG 2 AA** is correct

**Comment by ciar4n** [comment]
@coolcat-creations @svom Hover state for dropdown items?  I can't seem to find them in https://www.dropbox.com/s/jlgncxdggip22rh/Joomla4-031217-lj-white.pdf?dl=0

**Comment by brianteeman** [comment]
@C-Lodder you are saying these colours match AA or that they should match AA?

**Comment by C-Lodder** [comment]
I'm saying they should, and apparent they already do:

> as my a11y checker is saying that the contrast between foreground and background colors meets WCAG 2 AA contrast ratio thresholds

**Comment by svom** [comment]
@ciar4n Do you mean this part of the design, or the hover state of the button itself?

If the buttons look to big (i guess you'll have a feeling for this, i would reduce the vertical padding about 2px each side. We're not fixed about this (or at least me), it's all about iteration.

image

**Comment by C-Lodder** [comment]
Had to grab the hex from the screenshot as I'm at work now, but according to the colour contrast checker, the current green BG will only pass AA with white text if the text is over 18.66px which I assume it isn't.

@coolcat-creations Have the a11y team signed off on this?

**Comment by svom** [comment]
Actually it should pass AA with small text too

bildschirmfoto 2018-02-20 um 10 41 33

**Comment by ciar4n** [comment]
@svom I amended the padding as suggested. Also added the shadow in your screenshot.

image

What's left is background color, font and icons which will be separate PRs.

**Comment by ciar4n** [comment]
> Do you mean this part of the design, or the hover state of the button itself?

Just the state on hover. So when the dropdown menu items are hover? Eg. color background with white text, greyed background etc.

**Comment by brianteeman** [comment]
@svom thats not the colour being used. The green being used is #28a745 and #218838 on hover

it is only the onhover colour that passes

Not sure what the other colours are supposed to be but the yellow fails on both states

**Comment by brianteeman** [comment]
Not sure if this is out of scope or not but the toggle here needs some text for a11y - perhaps an aria-label="more options" ?
chrome_2018-02-20_09-01-53

**Comment by svom** [comment]
@brianteeman 

Where do you get that colour codes from?

Regarding the sketch-document there are the following colourcodes for the buttons (or in general):

Green:
#2F7D32 (Ratio: 5.1:1)

Blue:
#0377BE (Ratio 4.8:1)

Darkblue:
#1A466B (Ratio: 9.8:1)

Red: 
#C52827 (Ratio 5.7:1)

Grey (for Example for locked articles):
#696969 (Ratio: 5.5:1)

Menu-BG-Blue:
#006FB5 (Ratio: 5.3:1)

For the yellow colour you are right, but i guess we need to switch to the light blue here to achieve the needed colour contrast. I tried out a lot of different variations but it's nearly impossible to get a nice yellow fitting the contrast to white with wcag 2 AA.

**Comment by brianteeman** [comment]
I got the color codes from the css

**Comment by svom** [comment]
Oh okay. :/  So these are the colours we used in the sketch document, guess you need a more detailed overview of all the colours used, right?

**Comment by svom** [comment]
@ciar4n I will try to deliver the hover states during the next days, if that's fine. Just working on the custom elements at the moment

**Comment by brianteeman** [comment]
The colors being used are here https://github.com/ciar4n/40-backend-template/blob/0ac2edd69d1e7e1a17f49449a5eacd3048aee284/administrator/templates/atum/css/bootstrap.css#L2090

**Comment by C-Lodder** [comment]
@svom I asked on Skype a couple of days ago for the colour palette ;)

**Comment by ciar4n** [comment]
Dropdown styling add...

image

**Comment by ciar4n** [comment]
Spacer removed...

image

**Pull request merged**